stdlib/ysh/args.ysh

```diff
@@ -80,26 +79,46 @@ proc parser (; place ; ; block_def) {
   call place->setValue(p)
 }
 
-proc flag (short, long ; type='bool' ; default=null, help=null) {
+const kValidTypes = [Bool, Float, Int, Str]
+const kValidTypeNames = []
+for vt in (kValidTypes) {
+  call kValidTypeNames->append(vt.name)
+}
+
+func isValidType (type) {
+  try {
+    for valid in (kValidTypes) {
+      if (type is valid) {
+        return (true)
+      }
+    }
+  }
+  return (false)
+}
+
+proc flag (short, long ; type=Bool ; default=null, help=null) {
   ## Declare a flag within an `arg-parse`.
   ##
   ## Examples:
   ##
   ##   arg-parse (&spec) {
   ##     flag -v --verbose
-  ##     flag -n --count ('int', default=1)
-  ##     flag -f --file ('str', help="File to process")
+  ##     flag -n --count (Int, default=1)
+  ##     flag -p --percent (Float, default=0.0)
+  ##     flag -f --file (Str, help="File to process")
   ##   }
 
-  # bool has a default of false, not null
-  if (type === 'bool' and default === null) {
-    setvar default = false
+  if (type !== null and not isValidType(type)) {
+    var type_names = ([null] ++ kValidTypeNames) => join(', ')
+    error "Expected flag type to be one of: $type_names" (code=2)
   }
 
-  # TODO: validate `type`
+  # Bool has a default of false, not null
+  if (type is Bool and default === null) {
+    setvar default = false
+  }
 
-  # TODO: Should use "trimPrefix"
-  var name = long[2:]
+  var name = long => trimStart('--')
 
   ctx emit flags ({short, long, name, type, default, help})
```

Comment:
Actually I also realized it is possible for us to store `type.name` in the "spec", so we have a string rather than a type object ... though maybe that doesn't actually make anything easier